File: frontend/src/api/projects.js
import { apiRequest } from "./client";

export function listProjects() {
  return apiRequest("/api/projects");
}

export function createProject({ projectName, description }) {
  return apiRequest("/api/projects", {
    method: "POST",
    body: { project_name: projectName, description }
  });
}

export function regenerateProjectKey(projectId, { keyName } = {}) {
  return apiRequest(`/api/projects/${encodeURIComponent(projectId)}/keys`, {
    method: "POST",
    body: { key_name: keyName }
  });
}

File: backend/app/routes/projects.py
from typing import Annotated

from fastapi import APIRouter, Depends

from backend.app.dependencies import current_user
from backend.app.services.project_service import ProjectService
from backend.models.projects import CreateProjectRequest, GenerateProjectKeyRequest

router = APIRouter(prefix="/api/projects", tags=["projects"])


@router.get("")
def list_projects(user: Annotated[dict, Depends(current_user)]) -> list[dict]:
    return ProjectService().list_projects(user)


@router.post("")
def create_project(payload: CreateProjectRequest, user: Annotated[dict, Depends(current_user)]) -> dict:
    return ProjectService().create_project(user, payload.project_name, payload.description)


@router.post("/{project_id}/keys")
def regenerate_project_key(
    project_id: str,
    payload: GenerateProjectKeyRequest,
    user: Annotated[dict, Depends(current_user)],
) -> dict:
    return ProjectService().regenerate_key(user, project_id, payload.key_name)

